### ISSUE TITLE
"space-before-keywords" doesn't work nicelly with ES6's "super" keyword

### ISSUE THREAD

**Issue by mdziekon** [eslint/eslint#3946]
The following code throws with ``space-before-keywords: [2, "always"]`` enabled:

```javascript
(super.componentWillMount || noop)(...arguments);
```

Error thrown:
``Missing space before keyword "super". (space-before-keyword)``

I think ``super`` keyword in ES6 could be considered as a "normal" variable, so this rule kind of breaks on normal use cases.

ESLint version: ``v1.5.0``

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by nzakas** [comment]
Yeah, `super` should be treated like `this`

**Comment by alberto** [comment]
I can't get a failing test case.
This is passing in master:
```
{
  code: "class Foo { method() { super.bar }}",
  ecmaFeatures: { classes: true }
}
```

**Comment by alberto** [comment]
Ok, this one fails:
```
"class Foo { method() { (super.method)() }}"
```

**Comment by alberto** [comment]
@nzakas do you mean there shouldn't be any checks for super, like there aren't for this?

**Comment by nzakas** [comment]
Yup

**Comment by julien-f** [comment]
@nzakas I think the problem is more general (#3878 for instance), I think the rule should not applied when a keyword is used in place of an expression which can be the case for `super`, `yield` and `await` for example.

**Comment by nzakas** [comment]
Yeah, I think that's correct. Can you provide some test cases that fail right now?

**Comment by feross** [comment]
Any update on making this same fix for `yield` and `await` to fix https://github.com/eslint/eslint/issues/3878?
